### PR TITLE
Classes homework

### PR DESCRIPTION
**Pull request by Rehaeel**

### CODE REVIEW

File: 4.JavaScript-classes_constructor_new/script.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+class Vector {
+    constructor(arr) {
+        this.arr = arr;
+    }
+
+    toString() {
+        return `(${this.arr.toString()})`;
+    }
+
+    equals(secondArr) {
```
Comment (jbbch) [inline 4.JavaScript-classes_constructor_new/script.js:12]: Regarding your question about this one - that's exactly what it was supposed to do

File: 4.JavaScript-classes_constructor_new/script.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        if (this.arr.length === secondArr.arr.length) {
+            return true;
+        } else throw "Arrays length not equal";
```
Comment (jbbch) [inline 4.JavaScript-classes_constructor_new/script.js:15]: It's not a good idea to return true on one condition and to throw an error on the other. Method is not consistent. Unify returned types please

File: 4.JavaScript-classes_constructor_new/script.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    add(secondArr) {
+        return (
+            !this.equals(secondArr) ||
```
Comment (jbbch) [inline 4.JavaScript-classes_constructor_new/script.js:20]: I don't really like this. Logical OR returns first truthy value, but with your implementation if array are not equal, equals function throws an error, so your code will throw error in such case. Do you really need this?

Second - why do you return result of logical operation in add method? Add Function should do the work or not do the work, not return (true) or (false), right?

File: 4.JavaScript-classes_constructor_new/script.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            new Vector(this.arr.map((el, i) => el + secondArr.arr[i]))
+        );
+    }
+
+    subtract(secondArr) {
+        return (
+            !this.equals(secondArr) ||
```
Comment (jbbch) [inline 4.JavaScript-classes_constructor_new/script.js:27]: Same as above, fix everywhere please

File: 4.JavaScript-classes_constructor_new/script.js (hunk 2 above)
Comment (Rehaeel) [reply 4.JavaScript-classes_constructor_new/script.js:15]: I took the logic from the task. If arrays are equal, we return `true` to go further with a code. If not:
image

Comment (Rehaeel) [reply 4.JavaScript-classes_constructor_new/script.js]: But I understand, the code is less consistent. Changed to `return false`.

File: 4.JavaScript-classes_constructor_new/script.js (hunk 3 above)
Comment (Rehaeel) [reply 4.JavaScript-classes_constructor_new/script.js:20]: Second - from my understanding the task, it should return the Vector:
image
or throw an error:
image

File: 4.JavaScript-classes_constructor_new/script.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    constructor(arr) {
+        this.arr = arr;
+    }
+
+    toString() {
+        return `(${this.arr.toString()})`;
+    }
+
+    equals(secondArr) {
+        return this.arr.length === secondArr.arr.length ? true : false;
```
Comment (jbbch) [inline 4.JavaScript-classes_constructor_new/script.js:13]: Considering this.arr.length === secondArr.arr.length is an expression that evaluates to true or false, you can just return it

File: 4.JavaScript-classes_constructor_new/script.js (hunk 3 above)
Comment (Rehaeel) [reply 4.JavaScript-classes_constructor_new/script.js]: Deleted redundant `equals` check.
Done

**Pull request merged**